### PR TITLE
Prepare tests to phpunit10

### PR DESCRIPTION
**Pull request by Trukes**
The objective of this PR is to prepare for PHPUnit 10

### CODE REVIEW

**Comment by Trukes** [comment]
Still having 3 warnings:
1) Your XML configuration validates against a deprecated schema. Migrate your XML configuration using "--migrate-configuration"!

2) No tests found in class "Vich\UploaderBundle\Tests\Metadata\Driver\XmlDriverTest".

3) Class Vich\UploaderBundle\Tests\Storage\Flysystem\AbstractFlysystemStorageTest declared in /srv/vich-uploader/tests/Storage/Flysystem/AbstractFlysystemStorageTest.php is abstract

File: src/Handler/UploadHandler.php [hunk 1]
```diff
@@ -17,14 +17,15 @@
  *
  * @author Kévin Gomez <contact@kevingomez.fr>
  */
-final class UploadHandler extends AbstractHandler
+class UploadHandler extends AbstractHandler
```
Comment (garak) [inline src/Handler/UploadHandler.php:20]: We don't want to change the code to fit testing. Please revert any change about "final", as well as any other irrelevant change (I'll mark only the first one)

File: src/Handler/UploadHandler.php [hunk 2]
```diff
@@ -17,14 +17,15 @@
...
 {
     public function __construct(
         PropertyMappingFactory $factory,
         StorageInterface $storage,
         protected readonly FileInjectorInterface $injector,
         protected readonly EventDispatcherInterface $dispatcher
-    ) {
+    )
```
Comment (garak) [inline src/Handler/UploadHandler.php:27]: this is an example of code to revert

File: tests/Metadata/Driver/AnnotationDriverTest.php [hunk 3]
```diff
@@ -53,13 +53,13 @@ public function testReadUploadableAnnotation(): void
...
             ->method('getPropertyAnnotation')
-            ->willReturnCallback(static fn (ReflectionProperty $property): ?UploadableField => 'file' === $property->getName() ? new UploadableField('dummy_file', 'fileName') : null);
+            ->willReturnCallback(static fn(ReflectionProperty $property): ?UploadableField => 'file' === $property->getName() ? new UploadableField('dummy_file', 'fileName') : null);
 
         $driver = new AnnotationDriver($reader, [$this->managerRegistry]);
         $metadata = $driver->loadMetadataForClass(new \ReflectionClass($entity));
 
         self::assertInstanceOf(ClassMetadata::class, $metadata);
-        self::assertObjectHasAttribute('fields', $metadata);
+        self::assertIsArray($metadata->fields);
```
Comment (garak) [inline tests/Metadata/Driver/AnnotationDriverTest.php:62]: why?

**Review by garak — COMMENTED** [review]
The PHPUnit configuration file should be changed to be compatible with version 10

File: tests/Metadata/Driver/AnnotationDriverTest.php (hunk 3 above)
Comment (Trukes) [reply tests/Metadata/Driver/AnnotationDriverTest.php:62]: _assertObjectHasAttribute_ was deprecated:
https://github.com/sebastianbergmann/phpunit/issues/4601

Comment (IonBazan) [reply tests/Metadata/Driver/AnnotationDriverTest.php:62]: I think the import at the top of the file is invalid - `Metadata\ClassMetadata` should be replaced with `Vich\UploaderBundle\Metadata\ClassMetadata` - this way you can skip that assertion completely as `assertInstanceOf` will point to correct class now.

**Comment by IonBazan** [comment]
We can now use `#[DataProvider]`, `#[Group]`, `#[RequiresMethod]`, etc where applicable instead of the annotations.

**Comment by garak** [comment]
@Trukes any news on this?

**Comment by Trukes** [comment]
I will take some time to update this PR

File: phpunit.xml.dist [hunk 4]
```diff
@@ -1,34 +1,19 @@
...
-    </coverage>
-
-    <extensions>
-        <extension class="Vich\UploaderBundle\Tests\BypassFinalHook"/>
-    </extensions>
-
-    <listeners>
-        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener" />
-    </listeners>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.0/phpunit.xsd" bootstrap="vendor/autoload.php" colors="true" cacheDirectory=".phpunit.cache">
```
Comment (garak) [inline phpunit.xml.dist:2]: please keep changes at minimum in this file. Moreover, the configuration is detected as wrong by phpunit

File: phpunit.xml.dist.bak [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (garak) [inline phpunit.xml.dist.bak:1]: delete this file

File: src/Mapping/PropertyMapping.php [hunk 6]
```diff
@@ -13,7 +13,7 @@
  *
  * @author Dustin Dobervich <ddobervich@gmail.com>
  */
-final class PropertyMapping
+class PropertyMapping
```
Comment (garak) [inline src/Mapping/PropertyMapping.php:16]: revert this change

File: src/Mapping/PropertyMappingFactory.php [hunk 7]
```diff
@@ -15,7 +15,7 @@
  *
  * @internal
  */
-final class PropertyMappingFactory
+class PropertyMappingFactory
```
Comment (garak) [inline src/Mapping/PropertyMappingFactory.php:18]: revert this change

File: src/Metadata/MetadataReader.php [hunk 8]
```diff
@@ -14,7 +14,7 @@
  *
  * @internal
  */
-final class MetadataReader
+class MetadataReader
```
Comment (garak) [inline src/Metadata/MetadataReader.php:17]: revert this change

File: src/Storage/GaufretteStorage.php [hunk 9]
```diff
@@ -15,7 +15,7 @@
  *
  * @author Stefan Zerkalica <zerkalica@gmail.com>
  */
-final class GaufretteStorage extends AbstractStorage
+class GaufretteStorage extends AbstractStorage
```
Comment (garak) [inline src/Storage/GaufretteStorage.php:18]: revert this change

File: src/Templating/Helper/UploaderHelper.php [hunk 10]
```diff
@@ -9,7 +9,7 @@
  *
  * @author Dustin Dobervich <ddobervich@gmail.com>
  */
-final class UploaderHelper implements UploaderHelperInterface
+class UploaderHelper implements UploaderHelperInterface
```
Comment (garak) [inline src/Templating/Helper/UploaderHelper.php:12]: revert this change

File: tests/Form/Type/VichImageTypeTest.php [hunk 11]
```diff
@@ -156,7 +156,8 @@ public function testLiipImagineBundleIntegration(
         string $storageResolvedPath,
         string $imaginePattern,
         string $imagineResolvedPath
-    ): void {
+    ): void
```
Comment (garak) [inline tests/Form/Type/VichImageTypeTest.php:159]: revert this change

File: tests/Functional/UploadTest.php [hunk 12]
```diff
@@ -30,7 +30,7 @@ public function testFileIsUploadedWithFileType(string $uploadType, string $image
         $crawler = $client->followRedirect();
 
         self::assertTrue($client->getResponse()->isSuccessful());
-        self::assertFileExists($this->getUploadsDir($client).'/symfony_black_03.png', 'The file is uploaded');
+        self::assertFileExists($this->getUploadsDir($client) . '/symfony_black_03.png', 'The file is uploaded');
```
Comment (garak) [inline tests/Functional/UploadTest.php:33]: revert this change

File: tests/Functional/UploadTest.php [hunk 13]
```diff
@@ -42,7 +42,7 @@ public function testFileIsUploadedWithFileType(string $uploadType, string $image
             ],
         ]);
         self::assertTrue($client->getResponse()->isRedirect());
-        self::assertFileDoesNotExist($this->getUploadsDir($client).'/symfony_black_03.png', 'The file is deleted');
+        self::assertFileDoesNotExist($this->getUploadsDir($client) . '/symfony_black_03.png', 'The file is deleted');
```
Comment (garak) [inline tests/Functional/UploadTest.php:45]: revert this change

File: tests/Naming/PropertyDirectoryNamerTest.php [hunk 14]
```diff
@@ -22,9 +22,9 @@ public function fileDataProvider(): array
         $weirdEntity->someProperty = 'Yéô';
 
         return [
-            'plain' => ['foo',                       $entity,      'someProperty',     false],
-            'method call' => ['generated-file-name', $entity,      'generateFileName', false],
-            'translit.' => ['yeo',                   $weirdEntity, 'someProperty',     true],
+            'plain' => ['foo', $entity, 'someProperty', false],
```
Comment (garak) [inline tests/Naming/PropertyDirectoryNamerTest.php:25]: revert this change

File: tests/Naming/PropertyNamerTest.php [hunk 15]
```diff
@@ -39,7 +39,8 @@ public function testNameReturnsTheRightName(
         object $entity,
         string $propertyName,
         bool $transliterate
-    ): void {
+    ): void
```
Comment (garak) [inline tests/Naming/PropertyNamerTest.php:42]: revert this change

File: tests/Naming/PropertyNamerTest.php [hunk 16]
```diff
@@ -102,4 +103,4 @@ public function testConfigurationFailsIfThePropertyIsntSpecified(): void
 
         $namer->configure(['incorrect' => 'options']);
     }
-}
+}
```
Comment (garak) [inline tests/Naming/PropertyNamerTest.php:106]: revert this change

File: tests/Naming/SmartUniqidNamerTest.php [hunk 17]
```diff
@@ -7,18 +7,20 @@
...
             // case -> original name, result pattern
             'typical' => ['lala.jpeg', '/lala-[[:xdigit:]]{22}\.jpeg/'],
             'accented' => ['làlà.mp3', '/lala-[[:xdigit:]]{22}\.mp3/'],
             'spaced' => ['a Foo Bar.txt', '/a-foo-bar-[[:xdigit:]]{22}\.txt/'],
             'special char' => ['yezz!.png', '/yezz-[[:xdigit:]]{22}\.png/'],
-            'long basename' => [\str_repeat('a', 256).'.txt', '/a{228}-[[:xdigit:]]{22}\.txt/'],
-            'long extension' => ['a.'.\str_repeat('a', 256), '/a-[[:xdigit:]]{22}\.a{230}/'],
-            'long basename and extension' => [\str_repeat('a', 256).'.txt'.\str_repeat('a', 256),
-                                              '/a{228}-[[:xdigit:]]{22}\.txt/', ],
+            'long basename' => [\str_repeat('a', 256) . '.txt', '/a{228}-[[:xdigit:]]{22}\.txt/'],
```
Comment (garak) [inline tests/Naming/SmartUniqidNamerTest.php:18]: revert this change

File: tests/Naming/SmartUniqidNamerTest.php [hunk 18]
```diff
@@ -51,4 +53,4 @@ public function testNameReturnsAnUniqueName(string $originalName, string $patter
 
         self::assertMatchesRegularExpression($pattern, $namer->name($entity, $mapping));
     }
-}
+}
```
Comment (garak) [inline tests/Naming/SmartUniqidNamerTest.php:56]: revert this change

File: tests/Naming/SubdirDirectoryNamerTest.php [hunk 19]
```diff
@@ -41,4 +41,4 @@ public function testNameReturnsTheRightName(string $fileName, string $expectedFi
 
         self::assertSame($expectedFileName, $namer->directoryName($entity, $mapping));
     }
-}
+}
```
Comment (garak) [inline tests/Naming/SubdirDirectoryNamerTest.php:44]: revert this change

File: tests/Naming/UniqidNamerTest.php [hunk 20]
```diff
@@ -12,17 +12,17 @@
...
         return [
             // original_name, guessed_extension, pattern
-            ['lala.jpeg',     null,              '/[a-z0-9]{13}.jpeg/'],
-            ['lala.mp3',      'mpga',            '/[a-z0-9]{13}.mp3/'],
-            ['lala',          'mpga',            '/[a-z0-9]{13}.mpga/'],
-            ['lala',          null,              '/[a-z0-9]{13}/'],
-            ['lala.0',        null,              '/[a-z0-9]{13}\\.0/'],
-            ['lala.data.0',   null,              '/[a-z0-9]{13}\\.0/'],
-            ['lala.data.0',   'gzip',            '/[a-z0-9]{13}\\.0/'],
+            ['lala.jpeg', null, '/[a-z0-9]{13}.jpeg/'],
```
Comment (garak) [inline tests/Naming/UniqidNamerTest.php:19]: revert this change

File: tests/Naming/UniqidNamerTest.php [hunk 21]
```diff
@@ -51,4 +51,4 @@ public function testNameReturnsAnUniqueName(string $originalName, ?string $guess
 
         self::assertMatchesRegularExpression($pattern, $namer->name($entity, $mapping));
     }
-}
+}
```
Comment (garak) [inline tests/Naming/UniqidNamerTest.php:54]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 22]
```diff
@@ -68,7 +68,7 @@ public function testRemove(): void
             ->willReturn('test.txt');
 
         $this->storage->remove($this->object, $this->mapping);
-        self::assertFalse($this->root->hasChild('uploads'.\DIRECTORY_SEPARATOR.'test.txt'));
+        self::assertFalse($this->root->hasChild('uploads' . \DIRECTORY_SEPARATOR . 'test.txt'));
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:71]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 23]
```diff
@@ -218,7 +218,7 @@ public function testResolveStream(): void
         $this->mapping
             ->expects(self::once())
             ->method('getUploadDestination')
-            ->willReturn($this->root->url().'/uploads');
+            ->willReturn($this->root->url() . '/uploads');
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:221]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 24]
```diff
@@ -264,8 +264,8 @@ public function resolveUriDataProvider(): array
      */
     public function testUploadedFileIsCorrectlyMoved(string $uploadDir, string $dir, string $expectedDir): void
     {
-        $uploadDir = $this->root->url().\DIRECTORY_SEPARATOR.$uploadDir;
-        $expectedDir = $this->root->url().\DIRECTORY_SEPARATOR.$expectedDir;
+        $uploadDir = $this->root->url() . \DIRECTORY_SEPARATOR . $uploadDir;
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:267]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 25]
```diff
@@ -314,7 +314,7 @@ public function testReplacingFileIsCorrectlyUploaded(): void
             ->willReturn('test.txt');
         $file
             ->method('getPathname')
-            ->willReturn($this->getValidUploadDir().'/test.txt');
+            ->willReturn($this->getValidUploadDir() . '/test.txt');
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:317]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 26]
```diff
@@ -325,7 +325,7 @@ public function testReplacingFileIsCorrectlyUploaded(): void
         $this->mapping
             ->expects(self::once())
             ->method('getUploadDestination')
-            ->willReturn($this->root->url().\DIRECTORY_SEPARATOR.'storage');
+            ->willReturn($this->root->url() . \DIRECTORY_SEPARATOR . 'storage');
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:328]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 27]
```diff
@@ -353,7 +353,7 @@ public function testReplacingFileWithDirectoryNamerIsCorrectlyUploaded(): void
             ->willReturn('test.txt');
         $file
             ->method('getPathname')
-            ->willReturn($this->getValidUploadDir().'/test.txt');
+            ->willReturn($this->getValidUploadDir() . '/test.txt');
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:356]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 28]
```diff
@@ -364,7 +364,7 @@ public function testReplacingFileWithDirectoryNamerIsCorrectlyUploaded(): void
         $this->mapping
             ->expects(self::once())
             ->method('getUploadDestination')
-            ->willReturn($this->root->url().\DIRECTORY_SEPARATOR.'storage');
+            ->willReturn($this->root->url() . \DIRECTORY_SEPARATOR . 'storage');
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:367]: revert this change

File: tests/Storage/FileSystemStorageTest.php [hunk 29]
```diff
@@ -402,4 +402,4 @@ public function filenameWithDirectoriesDataProvider(): array
             ],
         ];
     }
-}
+}
```
Comment (garak) [inline tests/Storage/FileSystemStorageTest.php:405]: revert this change

File: tests/Storage/Flysystem/AbstractFlysystemStorageTest.php [hunk 30]
```diff
@@ -61,7 +61,7 @@ public function testUpload(): void
         $file
             ->expects(self::once())
             ->method('getRealPath')
-            ->willReturn($this->root->url().\DIRECTORY_SEPARATOR.'uploads'.\DIRECTORY_SEPARATOR.'test.txt');
+            ->willReturn($this->root->url() . \DIRECTORY_SEPARATOR . 'uploads' . \DIRECTORY_SEPARATOR . 'test.txt');
```
Comment (garak) [inline tests/Storage/Flysystem/AbstractFlysystemStorageTest.php:64]: revert this change

File: tests/Storage/Flysystem/AbstractFlysystemStorageTest.php [hunk 31]
```diff
@@ -148,13 +148,13 @@ public function testResolvePath(?string $uploadDir, string $expectedPath, bool $
...
 
-    public function pathProvider(): array
+    public static function pathProvider(): array
     {
         return [
             //     dir,   path,                     relative
-            [null,  'file.txt',               true],
-            [null,  '/absolute/file.txt',     false],
-            ['foo', 'foo/file.txt',           true],
+            [null, 'file.txt', true],
```
Comment (garak) [inline tests/Storage/Flysystem/AbstractFlysystemStorageTest.php:155]: revert this change

File: tests/Storage/GaufretteStorageTest.php [hunk 32]
```diff
@@ -126,15 +126,15 @@ public function testResolveUriFileNull(): void
...
+    public static function pathProvider(): array
     {
         return [
             // protocol, fs identifier, upload dir, full path, relative
-            ['gaufrette', 'filesystemKey', null,   'gaufrette://filesystemKey/file.txt', false],
-            ['data',      'filesystemKey', null,   'data://filesystemKey/file.txt', false],
-            ['gaufrette', 'filesystemKey', 'foo',  'gaufrette://filesystemKey/foo/file.txt', false],
-            ['gaufrette', 'filesystemKey', null,   'file.txt', true],
-            ['gaufrette', 'filesystemKey', 'foo',  'foo/file.txt', true],
+            ['gaufrette', 'filesystemKey', null, 'gaufrette://filesystemKey/file.txt', false],
```
Comment (garak) [inline tests/Storage/GaufretteStorageTest.php:133]: revert this change

File: tests/Storage/GaufretteStorageTest.php [hunk 33]
```diff
@@ -211,7 +211,7 @@ public function testUploadSetsMetadataWhenUsingMetadataSupporterAdapter(): void
         $file
             ->expects(self::once())
             ->method('getPathname')
-            ->willReturn($this->getValidUploadDir().\DIRECTORY_SEPARATOR.'test.txt');
+            ->willReturn($this->getValidUploadDir() . \DIRECTORY_SEPARATOR . 'test.txt');
```
Comment (garak) [inline tests/Storage/GaufretteStorageTest.php:214]: revert this change

File: tests/Storage/GaufretteStorageTest.php [hunk 34]
```diff
@@ -265,7 +265,7 @@ public function testUploadDoesNotSetMetadataWhenUsingNonMetadataSupporterAdapter
...
```
Comment (garak) [inline tests/Storage/GaufretteStorageTest.php:268]: revert this change

File: tests/Storage/GaufretteStorageTest.php [hunk 35]
```diff
@@ -310,4 +310,4 @@ protected function getFilesystemMock(): Filesystem|MockObject
     {
         return $this->createMock(Filesystem::class);
     }
-}
+}
```
Comment (garak) [inline tests/Storage/GaufretteStorageTest.php:313]: revert this change

File: tests/Storage/StorageTestCase.php [hunk 36]
```diff
@@ -115,6 +115,6 @@ public function testResolveUriWithEmptyFile(?string $filename): void
 
     protected function getValidUploadDir(): string
     {
-        return $this->root->url().\DIRECTORY_SEPARATOR.'uploads';
+        return $this->root->url() . \DIRECTORY_SEPARATOR . 'uploads';
```
Comment (garak) [inline tests/Storage/StorageTestCase.php:118]: revert this change

File: tests/Storage/StorageTestCase.php [hunk 37]
```diff
@@ -115,6 +115,6 @@ public function testResolveUriWithEmptyFile(?string $filename): void
...
     }
-}
+}
```
Comment (garak) [inline tests/Storage/StorageTestCase.php:120]: revert this change

**Review by garak — COMMENTED** [review]
We need to change `@PHPUnit84Migration:risky` to `@PHPUnit100Migration:risky`

**Comment by garak** [comment]
I'm sorry but this is taking too much.
I'm closing in favour or #1389
